File: Modules/ServerTypes/Membership/MyProfileColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MyProfileRow } from "./MyProfileRow";

export interface MyProfileColumns {
    UserId: Column<MyProfileRow>;
    Username: Column<MyProfileRow>;
    DisplayName: Column<MyProfileRow>;
    FormInitials: Column<MyProfileRow>;
    Email: Column<MyProfileRow>;
    UserImage: Column<MyProfileRow>;
}

export class MyProfileColumns extends ColumnsBase<MyProfileRow> {
    static readonly columnsKey = 'Membership.MyProfile';
    static readonly Fields = fieldsProxy<MyProfileColumns>();
}

File: Modules/Membership/MyProfile/MyProfileColumns.cs
using Serenity.ComponentModel;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;

namespace WinAuthSample.Membership.Columns
{
    [ColumnsScript("Membership.MyProfile")]
    [BasedOnRow(typeof(MyProfileRow), CheckNames = true)]
    public class MyProfileColumns
    {
        [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
        public int UserId { get; set; }
       
        public string Username { get; set; }
        public string DisplayName { get; set; }
        [MaxLength(4), Size(4)]
        public string FormInitials { get; set; }
        [EmailAddressEditor]
        public string Email { get; set; }
  
      //   public string MobilePhoneNumber { get; set; }
        public string UserImage { get; set; }
      
       
    }
}